### PR TITLE
feat(pkcs11-tool): add new --undestroyable

### PR DESCRIPTION
**Pull request by AlexandreGonzalo**
Hi All,

In this PR, I am adding a new option which can be used to mark the certificates as undestroyable.
Please, let me know if you want this option for the pkcs11-tool.
Regards,
Alexandre.

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [ ] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

File: src/tools/pkcs11-tool.c [hunk 1]
```diff
@@ -272,6 +273,7 @@ static const struct option options[] = {
 	{ "private",		0, NULL,		OPT_PRIVATE },
 	{ "sensitive",		0, NULL,		OPT_SENSITIVE },
 	{ "extractable",	0, NULL,		OPT_EXTRACTABLE },
+	{ "undestroyable",	0, NULL,		OPT_UNDESTROYABLE },
```
Comment (Jakuje) [inline src/tools/pkcs11-tool.c:276]: This is missing a documentation in the manual page for the pkcs11-tool (hence the CI failure)

**Review by Jakuje — CHANGES_REQUESTED** [review]

File: src/tools/pkcs11-tool.c (hunk 1 above)
Comment (AlexandreGonzalo) [reply src/tools/pkcs11-tool.c:276]: Good catch!
Sorry for the mistake.

Comment (Jakuje) [reply src/tools/pkcs11-tool.c:276]: I meant the manual page in `doc/tools/pkcs11-tool.1.xml`.

**Comment by Jakuje** [comment]
Code-wise looks good. Can you squash the changes into single commit?

**Comment by AlexandreGonzalo** [comment]
> Code-wise looks good. Can you squash the changes into single commit?

Done

**Review by Jakuje — APPROVED** [review]

**Review by frankmorgner — APPROVED** [review]

**Comment by Jakuje** [comment]
Thank you for your contribution!

**Comment by zhangxiaohui1** [comment]
I noticed that the --undestroyable option is currently only implemented for certificate objects (when using --write-object with --type cert). However, it does not seem to be supported for other object types such as privkey, pubkey, secrkey, or data.

This behavior is not documented in the help text of --undestroyable, which gives the impression that it should work universally across all writable object types.

Could you clarify:

Is this limitation intentional for security or technical reasons?
Would the project be open to extending --undestroyable support to all object types?
This feature would be valuable for use cases where long-term key material (e.g., device identity keys) must be protected from accidental or malicious deletion.

Thank you for your work on this essential tool!

**Comment by Jakuje** [comment]
Sounds like this was the author original use case to have undestroyable certificate. I think it makes sense to support this for other object types so feel free to open a PR with the changes to do so!

**Comment by AlexandreGonzalo** [comment]
CKA_DESTROYABLE can be set for any storage object, it is not limited to certificates.
The limitation is because I could only test it with certificates.
Feel free to generalize it for all the storage objects.

**Pull request merged**